# Cuaderno Matematicas
Un cuaderno de matematicas escrito con Jupyter Notebooks. 
En este cuaderno encontraras formulas escrita en lenguaje matematico y **Ruby**.

A continuacion declarare unas `funciones` (funcionan como macros, o acciones repetidas) para facilitar la disposicion de resultados.

In [27]:
require "bigdecimal"

def display(info)  
  IRuby.display "#{info}", mime: 'text/html'
end

def numeroGrande(num)
  puts ("%.15f" % num).sub(/0*$/,"")
end

:numeroGrande

`display` Permite visualizar los resultados correctamente y con mas estilo.

### [TEMA 1] - Numeros Reales

Numeros racionales son aquellos que se pueden **Expresar como fraccion**

Ejemplo: $\frac{a}{b} \leftarrow$ Numero Racional ($\mathbb{Q}$)

Los Numeros Irracionales ($\mathbb{I}$) son aquellos que no se pueden expresar como fraccion.

Ejemplo: $\sqrt 2 \rightarrow 1.41421356237$

Su expresion decimal es **ilimitada** y **no periodica**.



$$
\fbox{
Los Numeros $Reales$ $\mathbb{R}$, es un grupo que contiene los $Irracionales$ $\mathbb{I}$ y $Racionales$ $\mathbb{Q}$ 
}$$

#### [Pagina 7] ~ Ejercicio 1
Calcula la expresion decimal de esots numeros fraccionarios e indica que tipo de numero es.

$a)$ $\frac{21}{45} \rightarrow$ $0.46... \rightarrow$ Entero ($\mathbb{Z}$)

$b)$ $\frac{52}{13} \rightarrow$ $4 \rightarrow$ Natural ($\mathbb{N}$)

### [Tema 1] - Aproximaciones Y Errores
A la hora de aproximar hay diferentes metodos, **por defecto, por exceso y redondeo**.

Ejemplo: 2.1739821

| Tipo | Por Defecto  | Por Exceso | Redondeo |
|--| -- | -- | -- |
|Decimas|2.1| 2.2| 2.2 |
| Centesimas |2.17|2.18|2.17 |  
|Milesimas | 2.173 | 2.174 | 2.174 |
|Diezmilesimas | 2.1739| 2.1740 | 2.1740 |

### [Tema 1] - Errores
Las aproximaciones causan **errores**, los cuales son la diferencia que hay del valor real al valor aproximado. Este sirve para saber la realidad de los datos.

Hay dos formulas para calcular el Error, **Formula del Error Absoluto** Y **La Formula ddel Error Relativo**.
<br/><br/>
<center><strong>Error Absoluto:</strong></center>

$$ ErrorAbsoluto = |ValorReal - ValorAproximado| $$

<br/><br/>
<center><strong>Error Relativo:</strong></center>

$$ ErrorRelativo = \frac{ErrorAbsoluto}{| ValorReal |} $$


**[Pagina 13] ~ Ejercicio 39**

Determina los errores absoluto y relativo cometidos al utilizar estas proximaciones.

$a) \frac{2}{3} \approx 0.67$

Aplicamos la formula del Error Absoluto:

$\frac{2}{3} = 0.66666$

$Error Absoluto = |0.66 - 0.67|$

In [3]:
preErrorAbsoluto = (2.0/3.0)-0.67
errorAbsoluto = preErrorAbsoluto.abs()
display "El Error Absoluto es: #{errorAbsoluto}"

"El Error Absoluto es: 0.0033333333333334103"

[NOTA]: La function `abs()` que se puede observar aplica que el numero sea positivo. (Absoluto)

Aplicamos la formula del Error Relativo.

$Error Relativo = \frac{ErrorAbsoluto}{0.66}$

Lo calculamos...

In [4]:
errorRelativo = errorAbsoluto / (2.0/3.0)
errorRelativoPorcentaje = (errorRelativo * 100).truncate(4)
display "El Error Relativo es de #{errorRelativoPorcentaje} %"

"El Error Relativo es de 0.5 %"

---

<br>
<br>

### [Tema 1] - Intervalos Y Semirectas
Un **Intevalo** es el conjunto de numeros reales comprendidos entre dos de ellos, llamados **extremos** del intervalo.


[Pagina 15] ~ Ejercicio 52

$a)(-7, -3)$

$b)(-\infty, 8)$

$c)(-10, -4)$

$d) (\infty, 1)$

------
[Pagina 15] ~ Ejercicio 53

$a) \{x \in \mathbb{R} \div -4 \leq x < 2\}$

$b) \{x \in \mathbb{R} \div x \geq 9 \} $

$c) \{x \in \mathbb{R} \div x < -5 \} $

$b) \{x \in \mathbb{R} -1 \leq x \leq 1 \} $

## Tema 2 - Potencias Y Logaritmos
### [Tema 2] - Potencias de exponente entero. Operaciones
Las potencias representan el numero de veces que un numero se multiplica a si mismo.

Ejemplos:

**Producto de potencias de la misma base o el mismo exponente**

A) $A^{\pm{c}} * A^{\pm{b}} = A^{c\pm{b}}$

B) $A^c * B^c \rightarrow (A\cdot B)^c$


**Cociente de potencias de la misma base o el mismo exponente**

A) $\frac{a^{n}}{a^{x}} \rightarrow a$

[Pagina 25] - Ejercicio 5

Expresa como potencia de base de un numero primo.

$A) \frac{1}{16} = 16 \rightarrow$

$
  \begin{array}{|l}
    \llap{2~~~~} 16 \\ \hline
    \llap{2~~~~} 8  \\ \hline
    \llap{2~~~~} 4  \\ \hline
    \llap{2~~~~} 2  \\ \hline
    \llap{1~~~~} 1   \\ \hline
  \end{array}
  $
  
$Resultado = 2^4 \rightarrow \frac{1}{2^4} \rightarrow 2^{-4}$


$B) 0.125 \rightarrow \frac{1000}{125}$

$$
  \begin{array}{|l}
    \llap{5~~~~} 25 \\ \hline
    \llap{5~~~~} 5  \\ \hline
    \llap{1~~~~} 1  \\ \hline
  \end{array}
  $$
  
  $$
  \begin{array}{|l}
    \llap{2~~~~} 500 \\ \hline
    \llap{2~~~~} 250  \\ \hline
    \llap{2~~~~} 125  \\ \hline
    \llap{2~~~~} 125  \\ \hline
    \llap{5~~~~} 125  \\ \hline
    \llap{5~~~~} 25  \\ \hline
    \llap{5~~~~} 5  \\ \hline
    \llap{1~~~~} 1  \\ \hline
  \end{array}
  $$

$Resultado = 5^3 \rightarrow \frac{2^5 * 5^3}{5^3} \rightarrow 2^5$

Se elimina el $5^3$ porque esta en los dos lados de la fraccion.

$B) 81 \rightarrow$

$
  \begin{array}{|l}
    \llap{3~~~~} 27 \\ \hline
    \llap{3~~~~} 9  \\ \hline
    \llap{3~~~~} 3  \\ \hline
    \llap{1~~~~} 1  \\ \hline
  \end{array}
  $

$Resultado = 3^3$




**[Pagina 25] ~ Ejercicio 7**

Escribe como potencia de exponente positivo.

$A) \frac{2}{5}^{-3} \cdot \frac{2}{5}^{5} \div \frac{2}{5}^{-4} \downarrow$

$\frac{2}{5}^{-3+5} \rightarrow \frac{2}{5}^{2} \div \frac{2}{5}^{-4} \rightarrow \frac{2}{5}^{2-(-4)} \rightarrow \frac{2}{5}^{2+4} \rightarrow \frac{2}{5}^{6}$



$B) (3^2)^5 \div 3^8 \cdot (\frac{1}{2})^5 \div (\frac{1}{2})^7$

Para resolver esto, tenemos primero que resolver cada una de las potencias por separado. (Eg: $(3^2)^5$ etc.. )

$(3^2)^5 = 3^{2 \cdot 5} \rightarrow 3^{10} \div 3^8 \rightarrow 3^{10-8} \rightarrow 3^2$

$(\frac{1}{2}^{5-7}) \rightarrow \frac{1}{2^2}$ | En este caso, como $1^{-2} = 1$ no hace falta añdirlo.

$3^2 * \frac{1}{2^{-2}}$

### [Tema 2] - Notacion Cientifica
La notacion cientifica es una forma de expresar **numeros muy grandes** y **muy pequeños** con facilidad. 

Ejemplos: 

In [63]:
numeroEjemplo = 1.3*10**(-9) #
numeroGrande(numeroEjemplo)

0.0000000013


Los calculos de arriba son equivalentes a:

$1.3 \cdot 10^{-9} \rightarrow 0.0000000013$

Porque: $10^{(-9)}$ = `10**(-9)`

**[Pagina 27] ~ Ejercicio 13**

Expresa en notacion cientifica.

$A) 540000 \rightarrow 5.4*10^{-4}$

$B) 0,0034 \rightarrow 3.4*10^{3}$

$C) -{6720000} \rightarrow 0.672 * 10^{-7}$


Demonstraciones:

In [49]:
a = 54000 * 10 ** -4
puts a.to_f

b = 0.0034 * 10 ** 3 
numeroGrande(b)

c = -6720000 * 10 ** -7
puts c.to_f


5.4
3.4
-0.672


[Pagina 27] - Ejercicio 22


$A) (2.3*10^{-6}) \div (3*10^{-6}) \rightarrow 2.3 \div 3 \rightarrow 7.6 * 10^{-1}$

In [62]:
a = (2.3 / 3).to_f
a2 = (6-6).to_i

puts "a es igual a; #{a}, y su exponente de 10 es #{a2}"

a es igual a; 0.7666666666666666, y su exponente de 10 es 0


$B) (4.1*10^{-3})^{2} \div (2*10^4) \rightarrow \frac{4.1}{2} \cdot \frac{10^{-6}}{10^4} \downarrow $

In [61]:
b = (4.1/2).to_f
b2 = (-6-4).to_i

puts "B es igual a #{b} y su exponente de 10 es #{b2}"

B es igual a 2.05 y su exponente de 10 es -10


$2.05 \rightarrow 2.05*10^{-10}$

### [Tema 2] - Radicales

**Radical** se conoce aquel número que no puede ser simplificado para extraer su raíz cuadrada o cúbica. Por ejemplo, √2, es decir, la raíz cuadrada de 2, vendría a ser 2, puesto no se puede simplificar más.

Ejemplos:

$\sqrt 27^{0.33333} \rightarrow 0.3333 ... = \frac{1}{3} \rightarrow 27^{\frac{1}{3}}$

In [ ]:
[Tema 2]